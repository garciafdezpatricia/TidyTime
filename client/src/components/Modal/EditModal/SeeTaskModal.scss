@use "../../../../styles/variables" as *;

.see-task-article {
    position: absolute;
	transform: translate(-50%, -50%);
	background-color: $white;
	border-radius: .4rem;
	box-sizing: border-box;
    z-index: 10;
    top: 50%;
	left: 30%;
	bottom: 50%;
	padding: 1rem 2rem;
	width: 35rem;
	height: 18rem;
    box-shadow: -.1rem .2rem 1.5rem 0 rgba(0,0,0,0.75);

    display: flex;
    flex-direction: column;
    gap: .8rem;
    font-size: .8rem;

    .see-task-header {
        display: flex;
        gap: 1rem;
        width: 100%;
        justify-content: space-between;

        .see-task-header-title {
            display: flex;
            align-items: center;
            justify-content: center;
            gap: .5rem;
            font-size: 1rem;
            background-color: $primary-blue;
            color: $white;
            padding-inline: .8rem;
            border-radius: .4rem;
        }

        .see-task-close-btn {
            border: none;
            width: fit-content;
            height: fit-content;
            padding: .3rem;
            border-radius: .4rem;
            background-color: $primary-grey;
        }
    }

    .see-task-title {
        font-size: 1rem;
        border: none;
        height: fit-content;
        justify-self: self-start;
        width: 100%;
        &:active, &:focus {
            border: none;
            outline: none;
            text-decoration: underline;
            text-decoration-color: $primary-green;
        }
    }

    .see-task-info {
        resize: none;
        padding-left: .2rem;
        border-radius: .4rem;
        border: none;
        background-color: #e6e6e66a;
        color: $black;
        &:active, &:focus {
            border: none;
            outline: none;
        }
    }

    .see-task-date {
        height: fit-content;
        padding-inline: .4rem;
        justify-self: self-start;
        width: fit-content;
    }
}