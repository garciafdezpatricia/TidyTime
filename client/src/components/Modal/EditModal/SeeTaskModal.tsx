import { useEffect, useState } from "react";
import { IoMdClose } from "react-icons/io";
import { useEventContext } from "../../Context/EventContext";
import { useClickAway } from "@uidotdev/usehooks";
import { FaCheckCircle } from "react-icons/fa";


export interface Props {
    onClose: (arg?:any) => void | any,
}

export default function SeeTaskModal({onClose} : Props) {

    const [title, setTitle] = useState("");
    const [desc, setDesc] = useState("");
    const [dueDate, setDueDate] = useState("");
    const { events, selectedEventId } = useEventContext();

    const ref = useClickAway(() => {
        onClose();
    })

    useEffect(() => {
        events.map((event) => {
            if (event.eventId === selectedEventId) {
                setTitle(event.title);
                setDesc(event.desc);
                const startDate = new Date(event.start.getTime() - (event.start.getTimezoneOffset() * 60000)).toISOString().slice(0, 16);
                setDueDate(startDate);
                return;
            }
        })
    }, [events, selectedEventId]);

    return (
        // @ts-ignore
        <article ref={ref} className="see-task-article">
            <section className="see-task-header">
                <p className="see-task-header-title"><FaCheckCircle /> Task</p>
                <button className="see-task-close-btn" onClick={() => onClose()}>
                    <IoMdClose size={"1.2rem"} />
                </button>
            </section>
            <input 
                placeholder="Title..." 
                className="see-task-title" 
                type="text" 
                value={title} 
                contentEditable={false}
            />
            <textarea 
                placeholder="Description..." 
                className="see-task-info" 
                value={desc}
                contentEditable={false}
            />
            <input 
                className="see-task-date" 
                type="datetime-local"
                value={dueDate}
                contentEditable={false}
            />
        </article>
    )
}