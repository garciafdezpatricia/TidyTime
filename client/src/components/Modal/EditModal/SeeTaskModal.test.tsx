import { render } from "@testing-library/react";
import '@testing-library/jest-dom';
import SeeTaskModal from '@/src/components/Modal/EditModal/SeeTaskModal';

describe('SeeTaskModal', () => {
    it('renders without crashing', () => {
        const { container } = render(
        <SeeTaskModal onClose={function (arg?: any) {
                throw new Error("Function not implemented.");
            } }/>);
        expect(container).toBeInTheDocument();
    });

    it('matches snapshot', () => {
        const { asFragment } = render(
        <SeeTaskModal onClose={function (arg?: any) {
                throw new Error("Function not implemented.");
            } }/>);
        expect(asFragment()).toMatchSnapshot();
      });
  });