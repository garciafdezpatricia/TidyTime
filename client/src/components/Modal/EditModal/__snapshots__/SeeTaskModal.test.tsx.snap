// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SeeTaskModal matches snapshot 1`] = `
<DocumentFragment>
  <article
    class="see-task-article"
  >
    <section
      class="see-task-header"
    >
      <p
        class="see-task-header-title"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M504 256c0 136.967-111.033 248-248 248S8 392.967 8 256 119.033 8 256 8s248 111.033 248 248zM227.314 387.314l184-184c6.248-6.248 6.248-16.379 0-22.627l-22.627-22.627c-6.248-6.249-16.379-6.249-22.628 0L216 308.118l-70.059-70.059c-6.248-6.248-16.379-6.248-22.628 0l-22.627 22.627c-6.248 6.248-6.248 16.379 0 22.627l104 104c6.249 6.249 16.379 6.249 22.628.001z"
          />
        </svg>
         Task
      </p>
      <button
        class="see-task-close-btn"
      >
        <svg
          fill="currentColor"
          height="1.2rem"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 512 512"
          width="1.2rem"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M405 136.798L375.202 107 256 226.202 136.798 107 107 136.798 226.202 256 107 375.202 136.798 405 256 285.798 375.202 405 405 375.202 285.798 256z"
          />
        </svg>
      </button>
    </section>
    <input
      class="see-task-title"
      contenteditable="false"
      placeholder="Title..."
      type="text"
      value=""
    />
    <textarea
      class="see-task-info"
      contenteditable="false"
      placeholder="Description..."
    />
    <input
      class="see-task-date"
      contenteditable="false"
      type="datetime-local"
      value=""
    />
  </article>
</DocumentFragment>
`;
